pkg/auth/jwt_auth_cache.go

```diff
@@ -0,0 +1,78 @@
+package auth
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/grpc-ecosystem/go-grpc-middleware/auth"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type jwtAuthCache struct {
+	headerScheme string
+	jwtSecret    []byte
+	tokenCache   map[string]time.Time
+	cacheOrder   []string
+	cacheSize    int
+}
+
+// NewJwtAuthCache creates an AuthCache that implements bearer style
+// JWT based authorization validation.
+func NewJwtAuthCache(secretFile string) (AuthCache, error) {
+	data, err := ioutil.ReadFile(secretFile)
```

Comment:
It may be best to let this function take a `[]byte` as an argument with the secret directly, as that would make unit testing significantly easier.